docs/codejail.rst

```diff
@@ -0,0 +1,43 @@
+Codejail service
+################
+
+The ``codejail`` devstack component (codejail-service) requires some additional configuration before it can be enabled. This page describes how to set it up and debug it.
+
+Background
+**********
+
+Both LMS and CMS can run Python code submitted by instructors and learners in order to implement custom Python-graded problems. By default this involves running the code on the same host as edxapp itself. Ordinarily this would be quite dangerous, but we use a sandboxing library called `codejail <https://github.com/openedx/codejail>`__ in order to confine the code execution in terms of disk and network access as well as memory, CPU, and other resource limits. Part of these restrictions are implemented via AppArmor, a utility available in some Linux distributions (including Debian and Ubuntu).
+
+While AppArmor provides good protection, a sandbox escape could still be possible due to misconfiguration or bugs in AppArmor. For defense in depth, we're setting up a dedicated `codejail service <https://github.com/openedx/codejail-service>`__ that will perform code execution for edxapp and which will allow further isolation.
+
+The default edxapp codejail defaults to unsafe, direct execution of Python code, and this remains true in devstack. We don't even have a way to run on-host codejail securely in devstack. In constrast, the codejail service refuses to run if codejail has not been configured properly, and we've included a way to run it in devstack.
```

Comment:
[opinion] I'd leave most of this for the openedx/codejail-service docs. It's going to change even more if the old behaviour is DEPRed (including removal). Here, I'd write more from the perspective that we have and want a codejail-service, and here is what is needed to work with it.

This is non-blocking. It is an opinion to minimize future doc updates as time passes.

Note: I could imagine an ADR that captures this in the codejail-service, if that doesn't already exist, and you could point to it if you feel it is worth it, without surrounding it with much redundant text.

Reply:
Yeah, I'll go ahead and trim this down, and save what I wrote for when I write the docs for the app repo.